src/DataService/NotificationListener.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Lullabot\Mpx\DataService;
+
+use Lullabot\Mpx\Service\AccessManagement\ResolveAllUrls;
+use Lullabot\Mpx\Service\AccessManagement\ResolveDomain;
+use Lullabot\Mpx\Service\IdentityManagement\UserSession;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Listen to events from MPX.
+ *
+ * @see https://docs.theplatform.com/help/wsf-subscribing-to-change-notifications
+ *
+ * @codeCoverageIgnore Until we refactor this to allow loading all objects.
+ */
+class NotificationListener
+{
+    /**
+     * The resolver for MPX services.
+     *
+     * @var ResolveDomain
+     */
+    protected $resolveDomain;
+
+    /**
+     * The notification URL.
+     *
+     * @var string
+     */
+    protected $uri;
+
+    /**
+     * The user session to use for authenticated requests.
+     *
+     * @var UserSession
+     */
+    protected $session;
+
+    /**
+     * The name of the service to listen to notifications on, such as 'Media Data Service.'.
+     *
+     * @var string
+     */
+    protected $service;
+
+    /**
+     * @var string
+     */
+    private $clientId;
+
+    /**
+     * NotificationListener constructor.
+     *
+     * @param string      $clientId A string to identify this client in debugging.
+     * @param UserSession $session  The user session to use for authenticated requests.
+     * @param string      $service  The name of the service to listen to notifications on, such as 'Media Data Service.'
```

Comment:
I guess the dot is needed, since you included it here again

Reply:
Now I know it's not, because I've seen this service name in a few places more :)